e2e_tests/test_prompt_gateway.py

```diff
@@ -260,3 +260,40 @@ def test_prompt_gateway_default_target(stream):
             response_json.get("choices")[0]["message"]["content"]
             == "I can help you with weather forecast or insurance claim details"
         )
+
+
+@pytest.mark.parametrize("prefill_enabled", [True, False])
+def test_prompt_gateway_arch_prefill(prefill_enabled):
+    body = {
+        "messages": [
+            {
+                "role": "user",
+                "content": "how is the weather",
+            }
+        ],
+        "prefill_enabled": prefill_enabled,
```

Comment:
this isn't right place as it would break the contract for openai. We should use overrides to configure prefill here
https://github.com/katanemo/arch/blob/main/demos/function_calling/arch_config.yaml#L14-L16

Reply:
This only for the prompt gateway